remove_withdrawn_participants.R

```diff
@@ -0,0 +1,96 @@
+
+# Functions ---------------------------------------------------------------
+
+#' Get Identifier Variable Values for Withdrawn Participants
+#'
+#' This function gets the values of an identifier variable in 
+#' datasets that do not contain the ParticipantIdentifier variable, but 
+#' only for the corresponding ParticipantIdentifer values that are found 
+#' in a list containing ParticipantIdentifier values of withdrawn participants. 
+#' For example, if dataset A contains ParticipantIdentifier and DataKey columns, 
+#' and dataset B contains the DataKey column but not the ParticipantIdentifier 
+#' column, then we can map datasetB.DataKey to datasetA.ParticipantIdentifier 
+#' using datasetA.DataKey and datasetB.DataKey, filtering by values of 
+#' datasetA.ParticipantIdentifier that meet some criteria.
+#'
+#' @param dataset_name The name of a dataset that has the ParticipantIdentifier 
+#' variable and the mapping identifier variable (`mappingID_var`).
+#' @param mappingID_var The name of an identifier variable that can be 
+#' mapped to ParticipantIdentifier in the `dataset_name`'s parent dataset.
+#'
+#' @return The values of `mappingID_var` after filtering the dataset.
+#'
+#' @examples
+#' values_to_withdraw <- 
+#' get_mappingID_vals_to_withdraw("my_dataset", "DataPointKey")
+#' 
+get_mappingID_vals_to_withdraw <- function(dataset_name, mappingID_var) {
+  arrow::open_dataset(paste0(AWS_PARQUET_DOWNLOAD_LOCATION, "/", dataset_name, "/")) %>% 
+    dplyr::select(dplyr::all_of(c("ParticipantIdentifier", mappingID_var))) %>% 
+    dplyr::filter(ParticipantIdentifier %in% participants_to_withdraw) %>% 
+    dplyr::collect() %>% 
+    dplyr::pull(mappingID_var) %>% 
+    unique()
+}
+
+# Main --------------------------------------------------------------------
+
+# Get list of ParticipantIdentifiers of withdrawn participants
+participants_to_withdraw <- 
+  arrow::open_dataset(paste0(AWS_PARQUET_DOWNLOAD_LOCATION, "/dataset_enrolledparticipants/")) %>% 
+  dplyr::select(ParticipantIdentifier, CustomFields_EOPRemoveData) %>% 
+  dplyr::filter(CustomFields_EOPRemoveData==1) %>%
```

Comment:
CustomFields_EOPRemoveData is a string. I can see empty strings ''. 

Change this to == '1'.

Reply:
Updated